src/mds/MDLog.cc

```diff
@@ -727,19 +738,11 @@ int MDLog::trim_all()
            << "/" << expiring_segments.size()
            << "/" << expired_segments.size() << dendl;
 
-  uint64_t last_seq = 0;
   if (!segments.empty()) {
-    last_seq = get_last_segment_seq();
-    if (!capped &&
-	!mds->mdcache->open_file_table.is_any_committing() &&
-	last_seq > mds->mdcache->open_file_table.get_committing_log_seq()) {
-      submit_mutex.unlock();
-      mds->mdcache->open_file_table.commit(new C_OFT_Committed(this, last_seq),
-					   last_seq, CEPH_MSG_PRIO_DEFAULT);
-      submit_mutex.lock();
-    }
+    try_to_commit_open_file_table();
   }
 
+  uint64_t last_seq = get_last_segment_seq();
```

Comment:
This would assert when `segments` is empty. How about passing `last_seq` to `try_to_commit_open_file_table()`

Reply:
Sounds good and will fix it.